### PR TITLE
Feature/get bitbucket user

### PR DESCRIPTION
**Pull request by decoyjoe**
* Update init script to parameterize Bitbucket version with default `5.2.3`.
* Created `Get-BBServerUser` function for getting a Bitbucket Server user account.
* Added a `Parameter` parameter to `Invoke-BBServerRestMethod` which takes a hashtable representing the request query parameters to include when calling an API resource.

### CODE REVIEW

File: BitbucketServerAutomation/Functions/Get-BBServerUser.ps1 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    #>
+    param(
+        [Parameter(Mandatory=$true)]
+        [object]
+        # An object that defines what Bitbucket Server to connect to and the credentials to use when connecting.
+        $Connection,
+
+        [string]
+        # Specifies a filter to only return users whose username, display name or email addresses contain the supplied filter string.
+        $Filter
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Get-BBServerUser.ps1:42]: Can you give some examples? Or is this any string and Bitbucket Server searches the whole record? Or link to the docs?

File: BitbucketServerAutomation/Functions/Invoke-BBServerRestMethod.ps1 [hunk 2]
```diff
@@ -68,6 +72,21 @@ function Invoke-BBServerRestMethod
...
+    if ($Parameter)
+    {
+        $requestQueryParameter = ''
+        foreach ($key in $Parameter.Keys)
+        {
+            $requestQueryParameter += '&{0}={1}' -f $key, $Parameter[$key]
+        }
+
+        $requestQueryParameter = $requestQueryParameter.TrimStart('&')
+        $requestQueryParameter = [Uri]::EscapeUriString($requestQueryParameter)
```
Comment (splatteredbits) [inline BitbucketServerAutomation/Functions/Invoke-BBServerRestMethod.ps1]: `EscapeUriString` doesn't escape `&` or `=` characters, which would let users inject arbitrary parameters. Instead, use `EscapeDataString` on the query string names and values:

```
$requestQueryParameter += '&{0}={1}' -f [uri]::EscapeDataString($key), [uri]::EscapeDataString($Parameter[$key])
```

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: BitbucketServerAutomation/Functions/Invoke-BBServerRestMethod.ps1 (hunk 2 above)
Comment (decoyjoe) [reply BitbucketServerAutomation/Functions/Invoke-BBServerRestMethod.ps1]: Oh good idea, I'll make that change.

File: BitbucketServerAutomation/Functions/Get-BBServerUser.ps1 (hunk 1 above)
Comment (decoyjoe) [reply BitbucketServerAutomation/Functions/Get-BBServerUser.ps1:42]: The filter is just a simple string of characters that must appear in either the user's username, display name, or email address in Bitbucket.

I can use the verbiage straight [from the docs](https://docs.atlassian.com/bitbucket-server/rest/5.2.0/bitbucket-rest.html#idm46256608217776). What do you think of this?
```
[string]
# If specified only users with usernames, display name or email addresses containing the supplied string will be returned.
$Filter
```

Comment (splatteredbits) [reply BitbucketServerAutomation/Functions/Get-BBServerUser.ps1:42]: I would say `Filter to use to find a specific user. This is passed to Bitbucket Server as-is. It searches a user's username, display name, and email address for the filter. If any field matches, that user is returned.`

**Comment by decoyjoe** [comment]
@splatteredbits Changes made, ready for another review.

**Review by splatteredbits — APPROVED** [review]

**Pull request merged**